src/ebonite/client/base.py

```diff
@@ -66,13 +75,17 @@ def push_model(self, model: Model, task: Task = None) -> Model:
         model = self.meta_repo.save_model(model)
         return model
 
-    def delete_model(self, model: Model, force=False):
+    def delete_model(self, model: Model, *, force=False, cascade=False):
         """
         Deletes :py:class:`~ebonite.core.objects.Model` instance from metadata and artifact repositories
-
         :param model: model instance to delete
         :param force: whether model artifacts' deletion errors should be ignored, default is false
+        :param cascade: whether should model be deleted with all asssociated images
+        :return: Nothing
         """
+        if cascade:
+            for image in model.images:
```

Comment:
I recommend using `self.meta_repo.get_images` here as `Model.images` is not guaranteed to reflect DB state.